packages/ethereum-contracts/contracts/interfaces/superfluid/ISuperTokenFactory.sol

```diff
@@ -76,6 +82,52 @@ interface ISuperTokenFactory {
         external
         returns (ISuperToken superToken);
 
+    /**
+     * @notice Creates a wrapper super token AND sets it in our canonical list OR reverts if it already exists
+     * @dev salt for create2 is the keccak256 hash of abi.encode(address(_underlyingToken))
+     * @param _underlyingToken Underlying ERC20 token
+     * @return ISuperToken the created supertoken
+     */
+    function createCanonicalERC20Wrapper(ERC20WithTokenInfo _underlyingToken)
+        external
+        returns (ISuperToken);
+
+    /**
+     * @notice Computes/Retrieves wrapper super token address given the underlying token address
+     * @dev We return from our canonical list if it already exists, otherwise we compute it
+     * @dev note that this function only computes addresses for SEMI_UPGRADABLE SuperTokens
+     * @param _underlyingToken Underlying ERC20 token address
+     * @return superTokenAddress Super token address
+     * @return isDeployed whether the super token is deployed AND set in our canonical mapping
+     */
+    function computeCanonicalERC20WrapperAddress(address _underlyingToken)
+        external
+        view
+        returns (address superTokenAddress, bool isDeployed);
+
+    /**
+     * @notice Gets the canonical ERC20 wrapper super token address given the underlying token address
+     * @dev We return the address if it exists and the zero address otherwise
+     * @param _underlyingTokenAddress Underlying ERC20 token address
+     * @return superTokenAddress Super token address
+     */
+    function getCanonicalERC20Wrapper(address _underlyingTokenAddress)
+        external
+        view
+        returns (address superTokenAddress);
+
+    /**
+     * @notice Initializes list of canonical wrapper super tokens.
+     * @dev Can only be set once by a hardcoded address
+     * @param _data an array of canonical wrappper super tokens to be set
+     */
+    function initializeCanonicalWrapperSuperTokens(
+        InitializeData[] calldata _data
+    ) external;
```

Comment:
I feel like we shouldn't add this to the public interface (same for the `InitializeData` struct).
@hellwolf do you agree?

Reply:
I guess since it's a throwaway only for us to call by creating our own transaction, this is probably not necessary here, going to remove it